src/macros.rs

```diff
@@ -40,7 +40,6 @@
 ///
 /// # fn main() -> Result<()> {
 /// #     let depth = 0;
-/// #     let err: &'static dyn std::error::Error = &ScienceError::RecursionLimitExceeded;
```

Comment:
I was looking at this for a minute trying to figure out why it was there... honestly still not really sure. I'm guessing it has something to do with asserting that `ScienceError` type implements error, but the `bail!` macro should do the same thing, sooooo.... Seems fine